conans/search/query_parse.py

```diff
@@ -0,0 +1,124 @@
+from conans.util.log import logger
```

Comment:
Big job here. Unfortunately, very hard to read and understand. 
I think it might fail badly (ugly exceptions instead of controlled ones), with inputs as negation ``not, !``, not equal ``!=``, etc.  Also as you point out, with double and single quotes. And users will certainly do things like that.

On the other hand, I have checked from some library or pre-build parser that could help, doesn't seem simple either. It can be left as-is by now I guess.